src/scripts/main.js

```diff
@@ -1 +1,80 @@
 'use strict';
+
+async function wait(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+const firstPromise = new Promise(async (resolve, reject) => {
+  let isTrue = false;
+
+  document.addEventListener('click', () => {
+    isTrue = true;
+    resolve('First promise was resolved');
+  });
+
+  await wait(3000);
+
+  if (!isTrue) {
+    reject(new Error('First promise was rejected'));
+  }
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  let isTrue = false;
+
+  document.addEventListener('click', () => {
+    resolve('Second promise was resolved');
+    isTrue = true;
+  });
+
+  if (isTrue !== true) {
+    document.addEventListener('contextmenu', () => {
+      resolve('Second promise was resolved');
+    });
+  }
```

Comment:
The if statement here will always be evaluated immediately after the Promise is created, and thus `isTrue` will always be `false`. The `contextmenu` event listener will always be added, regardless of whether the `click` event has fired or not. You should remove this if statement and always add the `contextmenu` event listener.